src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +427,69 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
```

Comment:
I thought about what you might be after @chinpei215 - "Roles" is something I always consider to be plural per natural person, that confused me

Is this what you are looking for?
Users: id, first_name (e.g. Mark, Tim)
Positions: id, title (e.g. Boss, Manager, Associate)
Teams: id, name (e.g. RD, QA, Support)
TeamsUsers: user_id, team_id, role_id aka TeamAssignments